From 9717338f1917fa5db63fe7c9758639417ab06f61 Mon Sep 17 00:00:00 2001
From: yangfl <yangfl@users.noreply.github.com>
Date: Mon, 4 Jan 2021 23:09:41 +0800
Subject: [PATCH 06/10] extmod: Use system mbedtls

---
 extmod/extmod.mk | 77 ++----------------------------------------------
 1 file changed, 2 insertions(+), 75 deletions(-)

--- a/extmod/extmod.mk
+++ b/extmod/extmod.mk
@@ -242,11 +242,8 @@ SRC_THIRDPARTY_C += $(addprefix $(AXTLS_
 	crypto/sha1.c \
 	)
 else ifeq ($(MICROPY_SSL_MBEDTLS),1)
-MBEDTLS_DIR = lib/mbedtls
-MBEDTLS_CONFIG_FILE ?= \"mbedtls/mbedtls_config_port.h\"
-GIT_SUBMODULES += $(MBEDTLS_DIR)
-CFLAGS_EXTMOD += -DMBEDTLS_CONFIG_FILE=$(MBEDTLS_CONFIG_FILE)
-CFLAGS_EXTMOD += -DMICROPY_SSL_MBEDTLS=1 -I$(TOP)/$(MBEDTLS_DIR)/include
+CFLAGS_EXTMOD += -DMICROPY_SSL_MBEDTLS=1
+LDFLAGS_MOD += -lmbedx509 -lmbedtls -lmbedcrypto
 ifeq ($(MICROPY_PY_SSL_ECDSA_SIGN_ALT),1)
 CFLAGS_EXTMOD += -DMICROPY_PY_SSL_ECDSA_SIGN_ALT=1
 LDFLAGS_EXTMOD += -Wl,--wrap=mbedtls_ecdsa_write_signature
